File: w04/d03/ASSIGNMENT/goblins_solution/goblins.js
// create goblins array
var goblins = [];
for (var g = 0; g < 100; g++) {
	goblins.push(false);
}

console.log('initial set of uncured goblins: ', goblins);

// outer loop that loop though all the goblins in the array 100 times
for (var i = 0; i < goblins.length; i++) {
	// inner loop initial start will equal i and will incrument as the loop above loops
	// each inner loop will also loop 100 times and everytime it loop, loop incrument by i + 1
	for (var j = i; j < goblins.length; j += (i + 1)) {
		//toggle the goblin's state
		if (!goblins[j]) {
			goblins[j] = true;
		} else {
			goblins[j] = false;
		}
	}
  // check each outer iteration
	console.log(goblins);
}

// change values in goblins array to numbers
var result = [];
for (var u = 0; u < goblins.length; u++) {	
	//if that goblin is true push that index into result
	if (goblins[u]) { 
		result.push(u + 1); 
	}
}

// The set of cured goblins is the set of perfect squares up to and including 100.
console.log('Cured goblins: ', result);

File: w04/d03/ASSIGNMENT/goblins_solution/goblins.rb
# create initial set of uncured goblins in an array
goblins = []
# loop though 100 times and push the goblins into that array
for sick in 0...100
	# using false to represent no cure and true to represent has cure
	# all goblins starts out with no cure
	goblins.push(false)
end

# outer loop, "meet" or loops 100 times as descirbed by the problem
for meet in 0...100
	# inner loop, start with the n-th goblin in line "n" being current time we meet with goblins 
	increment = meet
	until increment >= 100
		# during the meeting with each individual goblin we toggle their state
		if goblins[increment]
		    # takeath those who has
			goblins[increment] = false
		else
		    # giveath those who has not
			goblins[increment] = true
		end
		# we gonna increment as much as "n" but adding 1 because we started at 0
		increment = (increment + meet + 1)
	end
end

# find out who is still alive
cured = []
for gob in 0...100
    if goblins[gob]
        cured.push(gob+1)
    end
end
puts cured